client/web/src/marketing/toast/CodySurveyToast.tsx

```diff
@@ -40,25 +54,55 @@ const CodySurveyToastInner: React.FC<{ onSubmitEnd: () => void } & TelemetryProp
         setIsCodyForPersonalStuff(event.target.checked)
     }, [])
 
-    const [submitCodySurvey, { loading }] = useMutation<SubmitCodySurveyResult, SubmitCodySurveyVariables>(
-        SUBMIT_CODY_SURVEY,
+    const [submitCodySurvey] = useMutation<SubmitCodySurveyResult, SubmitCodySurveyVariables>(SUBMIT_CODY_SURVEY, {
```

Comment:
You can simplify the whole loading and error state handling here:

```suggestion
    const [submitCodySurvey, { loading, error }] = useMutation<SubmitCodySurveyResult, SubmitCodySurveyVariables>(SUBMIT_CODY_SURVEY, {
```

`useMutation` can return states like that.

Then you only need to use those states, you don't need to do the whole `setLoading` and `setShowError` dance

Reply:
Thanks @pjlast,
We initially opted for this approach, since we needed to monitor the loading/error states of `submitCodySurvey` and `updatePostSignupCompletion` mutations.

WDYT about the new update, where we use the loading/error state from both cases and didn't define a new state too.

```javascript
const [updatePostSignupCompletion, { loading: loadingPostSignup, error: setPostSignupError }] ...
const [submitCodySurvey, { loading: loadingCodySurvey, error: submitSurveyError }] ...

const loading = loadingCodySurvey || loadingPostSignup
const error = !!submitSurveyError || !!setPostSignupError
```